### PR TITLE
Add debug plot to spectrum analysis

### PR DESCRIPTION
**Pull request by joleroi**
Some debug plots for spectrum analysis. The are in ``spectrum.utils`` until I find a better place

TODO
* [ ] Add test on fill skymap

### CODE REVIEW

**Comment by joleroi** [comment]
@adonath I assigned you to review the way an event list is filled into a Skymap ``EventList.fill_skymap``. Do  you think this is a good way to do it or did you have something else in mind?

**Comment by joleroi** [comment]
@cdeil The remaining functions `fill_counts_image` and `fill_counts_header` seem redundant to me, what to you think? They are only used here

https://github.com/gammapy/gammapy/blob/94c3541d1b9ecb89184679f0435d153e5747899c/examples/wip_make_survey_map.py#L95

**Comment by cdeil** [comment]
+1 to remove old functionality if it has been superseded (e.g. by SkyMap here).
Honestly ... with all the recent changes, I've lost track a bit.

Wouldn't `sky_map.fill(events)` be nicer since it changes the `sky_map` object?

Please add a basic test if you change / introduce new functionality (`fill_skymap` in this case).

**Comment by joleroi** [comment]
debug_plot


Now I know why it's called a debug plot ... :grin:

File: gammapy/data/event_list.py [hunk 1]
```diff
@@ -354,6 +355,35 @@ def filter_circular_region(self, region):
...
+        Returns
+        -------
+        filled_map : `~gammapy.image.SkyMap`
+            Sky map with events filled
+        """
+        from gammapy.image import SkyMap
+
+        shape = skymap.data.shape
+
+        coords = skycoord_to_pixel(self.radec, skymap.wcs)
```
Comment (cdeil) [inline gammapy/data/event_list.py]: Filling counts images and cubes is implemented and pretty well tested.
See e.g. here:
https://github.com/gammapy/gammapy/blob/master/gammapy/scripts/tests/test_image_bin.py#L15

I'm +1 to use SkyMap and make better APIs, but you have to take the time and remove the other functions and update the tests.

**Comment by cdeil** [comment]
Rebase, please.

**Comment by cdeil** [comment]
@joleroi - Can you please finish up this PR for the 0.4 release?

Note that [SkyMap.fill](https://gammapy.readthedocs.org/en/latest/api/gammapy.image.SkyMap.html#gammapy.image.SkyMap.fill) now exists.

But getting a standard debug plot that shows on and reflected regions would be nice to have.
Of course, combining all useful info in one plot (regions, counts, exclusion mask, pointing positions) would be best, but it's also probably not worth spending too much time on this.

**Comment by cdeil** [comment]
I want to cut the Gammpy 0.4 release tonight and this is the last open PR.
@joleroi - OK to move to 0.5? Or do you want to finish this up and merge?

**Comment by cdeil** [comment]
Closing this PR without merging.

@joleroi - I think that's what we discussed on Slack? If no, feel free to re-open.

**Pull request closed without merging**